### DISCUSSION TITLE
Patchcore: the value of nearest neighbor size

### DISCUSSION THREAD

**Discussion by ghost** [open-edge-platform/anomalib#550] [Q&A] [answered]
Thanks to your efforts,
I've noticed in the code of patchcore doing k nearest neighbor search,
the value of k is set to 9,
can I ask how to determine the value of k?
can I set k=1?

**Comment by samet-akcay** [comment, marked as answer]
@clarezei, this would be application-dependent, but in our implemenation we tune this based on the mvtec benchmarks. You could potentially use our hyper-parameter search [tool](https://github.com/openvinotoolkit/anomalib/tree/main/tools/hpo).
